### PR TITLE
Feature/cli iexec app use voucher

### PR DESCRIPTION
**Pull request by abbesBenayache**

### CODE REVIEW

File: test/bellecour-fork/voucher-config.js [hunk 1]
```diff
@@ -1 +1 @@
-export const VOUCHER_HUB_ADDRESS = 'TO BE DEPLOYED';
+export const VOUCHER_HUB_ADDRESS = '0xfaAddC93baf78e89DCf37bA67943E1bE8F37Bb8c';
```
Comment (cedric25) [inline test/bellecour-fork/voucher-config.js:1]: Is it okay to commit this or is it better to keep it to 'TO BE DEPLOYED'?
Or is it the definitive voucher hub address?

File: src/cli/cmd/iexec-app.js [hunk 2]
```diff
@@ -1038,27 +1040,23 @@ run
...
       spinner.stop();
 
       if (!opts.force) {
         await prompt.custom(
-          `Do you want to spend ${formatRLC(
-            totalCost,
-          )} RLC to execute the following request: ${pretty({
+          `Do you want to spend ${formatRLC(totalCost)} nRLC ${
+            sponsored > 0 ? `(${sponsored} nRLC sponsored by the voucher)` : ''
+          } to execute the following request: ${pretty({
```
Comment (cedric25) [inline src/cli/cmd/iexec-app.js:1059]: Before `totalCost` was expressed in RLC, now it is nRLC?
Or should it remain as RLC in this sentence?

```suggestion
          `Do you want to spend ${formatRLC(totalCost)} RLC ${
            sponsored > 0 ? `(${sponsored} RLC sponsored by the voucher)` : ''
          } to execute the following request: ${pretty({
```

File: src/cli/cmd/iexec-app.js [hunk 3]
```diff
@@ -1038,27 +1040,23 @@ run
...
-
-      const { stake } = await checkBalance(chain.contracts, requester);
-      if (totalCost.gt(stake)) {
-        throw Error(
-          `Not enough RLC on your account (${formatRLC(
-            totalCost,
-          )} RLC required). Run "iexec account deposit" to topup your account.`,
-        );
-      }
+      const { total: totalCost, sponsored } = await estimateMatchOrders(
```
Comment (cedric25) [inline src/cli/cmd/iexec-app.js:1043]: Just noting that `estimateMatchOrders` will be performed twice:
 - Once here
 - And once again in the later called `matchOrders()`

Not ideal as some async work will be done twice, for nothing.

Would it be hard to change the `matchOrders()` signature and be able to give it `totalCost` and `sponsored` parameters?
Maybe in a next PR?

**Review by cedric25 — COMMENTED** [review]
Some comments and questions.

Also, should we run `npm run generateCliDoc` in order for `CLI.md` to be up-to-date with the new `--use-voucher` parameter?

File: test/bellecour-fork/voucher-config.js (hunk 1 above)
Comment (abbesBenayache) [reply test/bellecour-fork/voucher-config.js:1]: the address should remain 'TO BE DEPLOYED' until the voucher is deployed on prod. it was just an error on my part, i accidentally committed the address I use for local tests. 

correction commit : 6134aac5a8a2e23537e2e0f58c8ba43e55c7356a

File: src/cli/cmd/iexec-app.js (hunk 2 above)
Comment (abbesBenayache) [reply src/cli/cmd/iexec-app.js:1059]: before, totalCost was calculated manually (I removed that code and replaced it with estimateMatchOrders), and now this function calculates both totalCost and sponsored in nRLC

File: src/cli/cmd/iexec-app.js (hunk 3 above)
Comment (abbesBenayache) [reply src/cli/cmd/iexec-app.js:1043]: yes, a refactoring of the matchOrders() function is definitely needed, but it won't be addressed in this PR

**Review by cedric25 — APPROVED** [review]

**Pull request merged**